File: in_Java/0841 Keys and Rooms.java
class Solution {
    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        int n = rooms.size();
        int[] visited = new int[n];
        Stack<Integer> st = new Stack<>();
        int cur;
        st.push(0);               
        while(!st.isEmpty()){
            cur = st.pop();
            visited[cur] = 1;
            for(int j : rooms.get(cur)){
                if(visited[j] == 0){
                    st.push(j);
                }
            }
        }
        int res = 0;
        for(int i : visited) res += i;
        return res == n;
    }
}

File: in_Python/0841 Keys and Rooms.py
class Solution(object):
    def canVisitAllRooms(self, rooms):
        """
        :type rooms: List[List[int]]
        :rtype: bool
        """
        N = len(rooms)
        visited = []
        q = []
        q.append(0)
        visited.append(0)
        while len(q) != 0:
            root = q.pop(0)
            nextnodes = rooms[root]
            for ni in nextnodes:
                if ni in visited:
                    continue
                else:
                    visited.append(ni)
                    q.append(ni)
        return len(visited) == N